### ISSUE TITLE
Loss not going down and renders look bad

### ISSUE THREAD

**Issue by dllu** [graphdeco-inria/gaussian-splatting#17]
hi I'm running it like so:

```
python train.py --ip 0.0.0.0 --port 13379 -s $colmap -m $outdir
python render.py -m $outdir
```

The logs from train.py look like this:

```
Optimizing /home/danlu/debug/gaussian-splatting-2023-07-10
Output folder: /home/danlu/debug/gaussian-splatting-2023-07-10 [11/07 13:56:37]
Reading camera 1130/1130 [11/07 13:56:39]
Loading Training Cameras [11/07 13:56:39]
Loading Test Cameras [11/07 13:56:52]
Number of points at initialisation :  532404 [11/07 13:56:52]
Training progress:  23%|████████████████▌                                                      | 7000/30000 [01:45<04:24, 86.86it/s, Loss=0.2195082]
[ITER 7000] Evaluating train: L1 0.2186211347579956 PSNR 12.067031860351562 [11/07 13:58:39]

[ITER 7000] Saving Gaussians [11/07 13:58:39]
Training progress: 100%|█████████████████████████████████████████████████████████████████████▉| 30000/30000 [06:19<00:00, 78.99it/s, Loss=0.2308751]

[ITER 30000] Saving Gaussians [11/07 14:03:11]

Training complete. [11/07 14:03:13]
```

But the output looks very bad, almost all the frames in `$outdir/ are like this:

![image](https://github.com/graphdeco-inria/gaussian-splatting/assets/14482624/969ac1b8-7030-4745-ab1f-7a2224094479)

or like this:
![image](https://github.com/graphdeco-inria/gaussian-splatting/assets/14482624/cba91697-752d-4a3d-a370-e6ead904d6c3)

and only a few frames (around 5 out of over 1000 frames) have any sort of recognizable structure in them.

![image](https://github.com/graphdeco-inria/gaussian-splatting/assets/14482624/284ce567-4e8e-4b2c-912f-b89d5627d368)

On the other hand, using [taichi gaussian splatting](https://github.com/wanmeihuali/taichi_3d_gaussian_splatting) with the exact same dataset, the output looks very reasonable.

Does anyone have any ideas for the parameters that I might have to change?

**Comment by Snosixtyboo** [comment]
Hi, would it be possible to get access to the dataset (your colmap input) so we can get an idea what's going on?

**Comment by dllu** [comment]
Unfortunately the input dataset contains confidential proprietary data.

**Comment by Snosixtyboo** [comment]
That makes it hard... I would suggest reducing the scaling, initial, and final position learning rate, e.g., divide them all by a factor 3 for starters.

**Comment by Snosixtyboo** [comment]
> 40

We'd be happy to hear how it goes! There's a few things that could happen at this scale we did not observe, our evaluation was mainly room-sized or smaller outdoor areas. The fact your first run ended in 6 minutes on what seems like an urban data set seems curious.

**Comment by dllu** [comment]
> reducing learning rate

Thanks for the pro tip!

![image](https://github.com/graphdeco-inria/gaussian-splatting/assets/14482624/82a7c9ba-dd7e-46fd-bc8f-554659a63281)

Indeed, with the following parameters

```
--iterations 50000 --scaling_lr 0.0002 --rotation_lr 0.0002 --position_lr_init 0.00003 --opacity_lr 0.01 --feature_lr 0.001
```

it seems the loss is decreasing healthily. Let's see what it ends up at after 50k iterations!

**Comment by dllu** [comment]
OK, after 50k iterations with the above settings, it is looking pretty reasonable.

![image](https://github.com/graphdeco-inria/gaussian-splatting/assets/14482624/6b1ff981-1e7a-4edb-a53c-1715e7486547)

![image](https://github.com/graphdeco-inria/gaussian-splatting/assets/14482624/97b84045-6040-49e1-9eb9-7c257dbedbad)

**Comment by grgkopanas** [comment]
How is it number-wise compared to the taichi implementation? Do you think there is room for improvement considering the hyper-params?

**Comment by Snosixtyboo** [comment]
> OK, after 50k iterations with the above settings, it is looking pretty reasonable.
> 
> ![image](https://user-images.githubusercontent.com/14482624/252793771-6b1ff981-1e7a-4edb-a53c-1715e7486547.png)
> 
> ![image](https://user-images.githubusercontent.com/14482624/252793809-97b84045-6040-49e1-9eb9-7c257dbedbad.png)

Great, thanks for letting us know!

A short word on urban data sets (there is also a section on it in the FAQ): The approach considers the distribution of the cameras and scales the learning rate accordingly (```spatial_lr_scale```). I.e., Gaussians will move faster if the cameras are 1,000 units apart than if it's only 1, since by default we have no absolute scaling information for how much one unit is. If cameras are distributed over kilometers or on a long straight path, you can therefore mess it up quite a bit, since Gaussians will zip off quickly. If you know the extent of your dataset, then extending the approach such that it sets learning rates from this information should be trivial. Or, if the coordinates of the point data sets are always at a known size (1unit = 1m), then replacing ```spatial_lr_scale``` by a constant should also do it.

**Comment by Snosixtyboo** [comment]
Follow-up: thanks a lot for bringing this to our attention! It turns out we overcomplicated things in the codebase and removing one adaptive parameter actually makes it work on small-scale AND large-scale data sets without problems. We will probably push this change today or tomorrow, but the implication would be that hyperparameter tuning for urban maps should either be unnecessary or much reduced.
